src/Interfaces.ts

```diff
@@ -85,10 +85,15 @@ export type IFieldIteratorFn = (
   fieldName: string,
 ) => void;
 
-export type NextResolverFn = () => Promise<any>;
+export type NextResolverFn<TSource> = (
+  source: TSource,
+  inputArgs: { [argName: string]: any },
+) => Promise<any>;
+
 export type DirectiveResolverFn<TSource, TContext> = (
-  next: NextResolverFn,
+  next: NextResolverFn<TSource>,
   source: TSource,
+  inputArgs: { [argName: string]: any },
```

Comment:
What's the difference between args and inputArgs?

Reply:
`args` is args of the directive. `inputArgs` is args of the field.
Maybe it's clearer to rename `args` to `directiveArgs`